### PR TITLE
Adapt to changes in podio

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
- Adapt to the changes in AIDASoft/podio#70 , add "includeSubfolder" option
- Adapt to the changes in AIDASoft/podio#69 , splitting of podio libraries
- CMake: Add logic to re-run class generator if edm4hep.yaml changes
ENDRELEASENOTES

### CODE REVIEW

**Comment by vvolkl** [comment]
Running this locally it seems that the code generator did not correctly update the include statements in the source files. Not sure yet though if this is due to a bug in the build system here or in podio.

**Comment by andresailer** [comment]
Did you update your local podio as well?

File: edm4hep/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,24 @@
...
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/edm4hep)
+
+
+# for this to work the command has to be in the same CMakeLists.txt as the PODIO_GENERATE_DICTIONARY
+# This ensures that we re-run the class generator if the yaml file is touched
+# the command is run of datalayout.yaml is newer then selection.xml
+add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/edm4hep/src/selection.xml  # we specify one output file so this doesn't always run
+  COMMENT "Re-Creating EDM4hep DataModel"
+  MAIN_DEPENDENCY ../edm4hep.yaml
+  # This removal ensures selection.xml is newer than datalayout.yaml, other wise this command always runs
```
Comment (andresailer) [inline edm4hep/CMakeLists.txt]:
```suggestion
  # This removal ensures selection.xml is newer than edm4hep.yaml, other wise this command always runs
```

File: edm4hep/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+# we need to boostrap the data model, so this has to be executed in the cmake run
+execute_process(
+    COMMAND ${CMAKE_COMMAND} -E echo "Creating EDM4hep DataModel"
+    COMMAND python ${podio_DIR}/../../../python/podio_class_generator.py ../edm4hep.yaml ./ edm4hep
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/edm4hep)
+
+
+# for this to work the command has to be in the same CMakeLists.txt as the PODIO_GENERATE_DICTIONARY
+# This ensures that we re-run the class generator if the yaml file is touched
+# the command is run of datalayout.yaml is newer then selection.xml
```
Comment (andresailer) [inline edm4hep/CMakeLists.txt]:
```suggestion
# the command is run if edm4hep.yaml is newer than selection.xml
```

**Comment by vvolkl** [comment]
Yes, I ran the CI build script, but with the newer commits  this problem seems to have disappeared. There is another issue I see though: the podio generation seems to get triggered again halfway through the build: https://travis-ci.com/github/key4hep/EDM4hep/jobs/298479620#L651
I think this should be avoided?

**Pull request merged**